test/mock/rtps/PublisherHistory/fastdds/publisher/DataWriterHistory.hpp

```diff
@@ -284,6 +284,11 @@ class DataWriterHistory : public WriterHistory
         return false;
     }
 
+    void unacknowledged_sample_removed_functor(
```

Comment:
See previous comment regarding injecting the functor in the constructor

Reply:
Done in c8cde8cad